model: scikit: Data type of feature used for predict not respected

Predicted values are comming out as floats from regression models. We should
typecast to the feature's dtype.

Hi @pdxjohnny, can you please help me with the tests which are failing? `(model/scikit, 3.7)`, `(model/scikit, 3.8)` and `(model/xgboost, 3.8)`. The error shows 

> AssertionError: '0' not found in [-1, 0, 1, 2, 3, 4, 5, 6, 7]

Now, why would someone check for a string in an array of `int`s. Also, why is the error only in these 3 models?

--- a/model/scikit/dffml_model_scikit/scikit_base.py
+++ b/model/scikit/dffml_model_scikit/scikit_base.py
@@ -148,7 +148,10 @@ class ScikitContext(ModelContext):
             )
             target = self.parent.config.predict.name
             record.predicted(
-                target, self.clf.predict(predict)[0], self.confidence
+                target, 
+                self.parent.config.predict.dtype(self.clf.predict(predict)[0]) if \
+                    self.parent.config.predict.dtype is not str else \
+                    self.clf.predict(predict)[0], self.confidence
             )
             yield record
 
@@ -260,7 +263,10 @@ class ScikitContextUnsprvised(ScikitContext):
                 "Predicted cluster for {}: {}".format(predict, prediction)
             )
             target = self.parent.config.predict.name
-            record.predicted(target, prediction[0], self.confidence)
+            record.predicted(target,
+                self.parent.config.predict.dtype(prediction[0]) if \
+                    self.parent.config.predict.dtype is not str else \
+                    prediction[0], self.confidence)
             yield record
 
 

--- a/model/scikit/dffml_model_scikit/scikit_base.py
+++ b/model/scikit/dffml_model_scikit/scikit_base.py
@@ -148,10 +148,11 @@ class ScikitContext(ModelContext):
             )
             target = self.parent.config.predict.name
             record.predicted(
-                target, 
-                self.parent.config.predict.dtype(self.clf.predict(predict)[0]) if \
-                    self.parent.config.predict.dtype is not str else \
-                    self.clf.predict(predict)[0], self.confidence
+                target,
+                self.parent.config.predict.dtype(self.clf.predict(predict)[0])
+                if self.parent.config.predict.dtype is not str
+                else self.clf.predict(predict)[0],
+                self.confidence,
             )
             yield record
 
@@ -263,10 +264,13 @@ class ScikitContextUnsprvised(ScikitContext):
                 "Predicted cluster for {}: {}".format(predict, prediction)
             )
             target = self.parent.config.predict.name
-            record.predicted(target,
-                self.parent.config.predict.dtype(prediction[0]) if \
-                    self.parent.config.predict.dtype is not str else \
-                    prediction[0], self.confidence)
+            record.predicted(
+                target,
+                self.parent.config.predict.dtype(prediction[0])
+                if self.parent.config.predict.dtype is not str
+                else prediction[0],
+                self.confidence,
+            )
             yield record
 
 

